databricks_cli/stack/api.py

```diff
@@ -0,0 +1,340 @@
+        template.
+
+        :param resource: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE and
+        RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :param resource_deploy_info: A dict of the resource's deployment info from the last
+        deployment. Will be None if this is the first deployment.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {...}}
+        :return: dict resource_deploy_info- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        # try:
+        resource_id = resource[RESOURCE_ID]
+        resource_service = resource[RESOURCE_SERVICE]
+        resource_properties = resource[RESOURCE_PROPERTIES]
+        # except KeyError as e:
+        #     raise StackError("%s doesn't exist in resource config" % str(e))
+
+        # Get physical ID for resource from last deployment
+        # physical_id = self._get_previous_resource_deploy_map(resource_id, resource_service)
+
+        physical_id = resource_deploy_info[RESOURCE_PHYSICAL_ID] if resource_deploy_info else None
+        if resource_service == JOBS_SERVICE:
+            physical_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource service '%s' not supported" % resource_service)
+
+        resource_deploy_info = {RESOURCE_ID: resource_id, RESOURCE_SERVICE: resource_service,
+                                RESOURCE_DEPLOY_TIMESTAMP: datetime.now(),
+                                RESOURCE_PHYSICAL_ID: physical_id,
+                                RESOURCE_DEPLOY_OUTPUT: deploy_output}
+        return resource_deploy_info
+
+    def validate_status(self, stack_status):
+        """
+        Validate fields within a stack status. This ensures that a stack status has the
+        necessary fields for stack deployment to function well.
+
+        If there is an error here, then it is either an implementation error that must be fixed by
+        a developer or the User edited the stack status file created by the program.
+
+        :param stack_status: dict- stack status that is created by the program.
+        :return: None. Raises errors to stop deployment if there is a problem.
+        """
+        if STACK_NAME not in stack_status:
+            raise StackError("'%s' not in status" % STACK_NAME)
+        if STACK_DEPLOYED not in stack_status:
+            raise StackError("'%s' not in status" % STACK_DEPLOYED)
+        for deployed_resource in stack_status[STACK_DEPLOYED]:
```

Comment:
stack_status also should have `STACK_RESOURCES` right? If so, please have a check there.

Reply:
yup, i'll add it